modules/swagger-codegen/src/main/resources/rust/api.mustache

```diff
@@ -39,26 +40,84 @@ impl<C: hyper::client::Connect>{{classname}} for {{classname}}Client<C> {
     fn {{{operationId}}}(&self, {{#allParams}}{{paramName}}: {{#isString}}&str{{/isString}}{{#isUuid}}&str{{/isUuid}}{{^isString}}{{^isUuid}}{{^isPrimitiveType}}{{^isContainer}}::models::{{/isContainer}}{{/isPrimitiveType}}{{{dataType}}}{{/isUuid}}{{/isString}}{{#hasMore}}, {{/hasMore}}{{/allParams}}) -> Box<Future<Item = {{^returnType}}(){{/returnType}}{{#returnType}}{{{.}}}{{/returnType}}, Error = Error<serde_json::Value>>> {
         let configuration: &configuration::Configuration<C> = self.configuration.borrow();
 
+{{#hasAuthMethods}}
+        let mut auth_headers = HashMap::<String, String>::new();
+        let mut auth_query = HashMap::<String, String>::new();
+{{#authMethods}}
+{{#isApiKey}}
+        match configuration.api_key {
+            None => (),
+            Some(ref apikey) => {
+                let key = apikey.key.clone();
+                let val = match apikey.prefix {
+                    Some(ref prefix) => format!("{} {}", prefix, key),
+                    None => key,
+                };
+            {{#isKeyInHeader}}
+                auth_headers.insert("{{keyParamName}}".to_owned(), val);
+            {{/isKeyInHeader}}
+            {{#isKeyInQuery}}
+                auth_query.insert("{{keyParamName}}".to_owned(), val);
+            {{/isKeyInQuery}}
```

Comment:
I assume `isKeyInHeader` and `isKeyInQuery` are guaranteed to be mutually exclusive?

Reply:
As I understand it yes, but these parameters aren't really well documented so I could be wrong.